src/julia-parser.scm

```diff
@@ -1271,7 +1271,11 @@
                 (syntax-deprecation s (string "abstract " (deparse spec))
                                     (string "abstract type " (deparse spec) " end")))
             (begin0 (list 'abstract spec)
-                    (if ty (expect-end s "abstract type"))))))
+                    (if ty (begin
+                      (let ((nt (peek-token s)))
+                        (if (or (newline? nt) (eqv? nt #\;))
+                            (take-token s)))
+                      (expect-end s "abstract type")))))))
```

Comment:
This indentation is a bit confusing. It makes it look like the `expect-end` call is in the "else" part of the `if`. I would use

```
(if ty
    (begin (let ((nt (peek-token s)))
             (if (or (newline? nt) (eqv? nt #\;))
                 (take-token s)))
           (expect-end s "abstract type")))
```

Or, you can remove the `begin`, since `let` has an implicit `begin` block:

```
(if ty
    (let ((nt (peek-token s)))
      (if (or (newline? nt) (eqv? nt #\;))
          (take-token s))
      (expect-end s "abstract type")))
```